plugins/tiddlywiki/tiddlyweb/tiddlywebadaptor.js

```diff
@@ -102,6 +103,53 @@ TiddlyWebAdaptor.prototype.getStatus = function(callback) {
 	});
 };
 
+TiddlyWebAdaptor.prototype.startSSE = function() {
+	// Get the mount point in case a path prefix is used
+	var host = this.getHost();
+	// Make sure it ends with a slash (it usually does)
+	if(host[host.length - 1] !== "/") {
+		host += "/";
+	}
+	// Setup the event listener
+	this.setupSSE(host);
+};
+function debounce(callback) {
+	var timeout = null;
+	return function() {
+		clearTimeout(timeout);
+		timeout = setTimeout(callback,$tw.syncer.throttleInterval);
+	};
+}
+
+TiddlyWebAdaptor.prototype.setupSSE = function (host, refresh) {
+	if (window.EventSource) {
+		var self = this;
+		if(this.eventsource && this.eventsource.readyState !== this.eventsource.CLOSED)
+			this.eventsource.close();
+		var source = this.eventsource = new EventSource(host + "events/plugins/tiddlywiki/tiddlyweb/wiki-change", {
+			withCredentials: true
+		});
+		var debouncedSync = debounce($tw.syncer.syncFromServer.bind($tw.syncer));
+		source.addEventListener("change", debouncedSync);
+		source.onerror = function () {
+			// set this to a no-op since we only want to sync on the first open
+			source.onopen = function(){};
+			// return if we're reconnecting because that's handled automatically
+			if (source.readyState === source.CONNECTING) return;
+			// wait for the errorRetryInterval
+			setTimeout(function () {
+				//call this function to set everything up again
+				self.setupSSE(host, true);
+			}, $tw.syncer.errorRetryInterval);
+		};
+		// call syncFromServer on open if necessary
+		if (refresh) source.onopen = function () {
+			//sync from server manually here to make sure we stay up to date
+			$tw.syncer.syncFromServer();
+		}
```

Comment:
I think this construction is very error-prone. The closing brace belongs to the `function() {`, but it looks like if it would belong to the if statement. IMO the missing braces for the if should be added.